lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -600,32 +573,15 @@ newTransactionLayer keyF networkId = TransactionLayer
                     pure keyhash'
 
             case cardanoTxIdeallyNoLaterThan era sealedTx of
-                InAnyCardanoEra ByronEra _ ->
-                    sealedTx
-                InAnyCardanoEra ShelleyEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver (const Nothing)
-                    Nothing (const Nothing) addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
-                InAnyCardanoEra AllegraEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver (const Nothing)
-                    Nothing (const Nothing) addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
-                InAnyCardanoEra MaryEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver policyResolver
-                    policyKeyM stakingScriptResolver addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
-                InAnyCardanoEra AlonzoEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver policyResolver
-                    policyKeyM stakingScriptResolver addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
-                InAnyCardanoEra BabbageEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver policyResolver
-                    policyKeyM stakingScriptResolver addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
-                InAnyCardanoEra ConwayEra (Cardano.Tx body wits) ->
-                    signTransaction keyF networkId witCountCtx acctResolver policyResolver
-                    policyKeyM stakingScriptResolver addressResolver inputResolver (body, wits)
-                    & sealedTxFromCardano'
+                InAnyCardanoEra txEra (Cardano.Tx body wits) ->
+                    case Write.toRecentEra txEra of
+                        Just recentTxEra -> Write.withConstraints recentTxEra $
+                            signTransaction
+                                keyF networkId witCountCtx acctResolver
+                                policyResolver policyKeyM stakingScriptResolver
+                                addressResolver inputResolver (body, wits)
+                            & sealedTxFromCardano'
+                        Nothing -> sealedTx
```

Comment:
Should probably error... It matters more now that we no longer support signing  `Shelley, ..., Alonzo` txs

Reply:
See: 8f70f9bf73ae5c8beb3261ea917818130ea8ba09